File: app/src/main/java/com/returnnotfound/stain/base/utils/DecimalFormatUtils.java
package com.returnnotfound.stain.base.utils;

import java.text.DecimalFormat;
import java.text.DecimalFormatSymbols;
import java.util.Locale;

public abstract class DecimalFormatUtils {
  private static DecimalFormat NUMBER_DECIMAL_FORMAT = new DecimalFormat(",##0.00", new DecimalFormatSymbols(Locale.ENGLISH));
  private static DecimalFormat NUMBER_DECIMAL_FORMAT_SIMPLE = new DecimalFormat("0.##", new DecimalFormatSymbols(Locale.ENGLISH));

  public static String formatPrice(double number) {
    if (number < 0) {
      number = 0;
    }
    return "$" + formatNumber(number);
  }

  public static String formatPriceSimple(double number) {
    if (number < 0) {
      number = 0;
    }
    return "$" + formatNumberSimple(number);
  }

  public static double getPriceFromPriceFormatted(String priceFormatted) {
    double result;
    if (priceFormatted.contains("$")) {
      int indexDollar = priceFormatted.indexOf("$");
      priceFormatted = priceFormatted.substring(0, indexDollar) + priceFormatted.substring(indexDollar + 1);
    }
    priceFormatted = priceFormatted.replaceAll("-", "");
    priceFormatted = priceFormatted.replaceAll(" ", "");
    priceFormatted = priceFormatted.replaceAll(String.valueOf(NUMBER_DECIMAL_FORMAT.getDecimalFormatSymbols().getGroupingSeparator()), "");
    try {
      result = Double.parseDouble(priceFormatted);
    } catch (Exception e) {
      result = 0d;
    }
    return result;
  }

  public static String formatNumber(double number) {
    return formatNumber(NUMBER_DECIMAL_FORMAT, roundNumber(number));
  }

  public static String formatNumberSimple(double number) {
    return formatNumber(NUMBER_DECIMAL_FORMAT_SIMPLE, roundNumber(number));
  }

  public static String formatNumber(DecimalFormat decimalFormat, double number) {
    return decimalFormat.format(number);
  }

  public static double roundNumber(float number) {
    return (Math.round(number * 100) / 100d);
  }

  public static double roundNumber(double number) {
    return (Math.round(number * 100) / 100d);
  }
}

File: app/src/main/java/com/returnnotfound/stain/base/utils/DecimalFormatUtils.kt
package com.returnnotfound.stain.base.utils

import java.text.DecimalFormat
import java.text.DecimalFormatSymbols
import java.util.*
import kotlin.math.roundToLong

object DecimalFormatUtils {
  private val NUMBER_DECIMAL_FORMAT = DecimalFormat(",##0.00", DecimalFormatSymbols(Locale.ENGLISH))
  private val NUMBER_DECIMAL_FORMAT_SIMPLE =
    DecimalFormat("0.##", DecimalFormatSymbols(Locale.ENGLISH))

  @JvmStatic
  fun formatPrice(number: Double): String {
    return "$" + formatNumber(if (number < 0) 0.0 else number)
  }

  @JvmStatic
  fun formatPriceSimple(number: Double): String {
    return "$" + formatNumberSimple(if (number < 0) 0.0 else number)
  }

  @JvmStatic
  fun getPriceFromPriceFormatted(price: String): Double {
    var priceFormatted = price
    val result: Double
    if (priceFormatted.contains("$")) {
      val indexDollar = priceFormatted.indexOf("$")
      priceFormatted =
        priceFormatted.substring(0, indexDollar) + priceFormatted.substring(indexDollar + 1)
    }
    priceFormatted = priceFormatted.replace("-".toRegex(), "")
    priceFormatted = priceFormatted.replace(" ".toRegex(), "")
    priceFormatted = priceFormatted.replace(
      NUMBER_DECIMAL_FORMAT.decimalFormatSymbols.groupingSeparator
        .toString().toRegex(), ""
    )
    result = try {
      priceFormatted.toDouble()
    } catch (e: Exception) {
      0.0
    }
    return result
  }

  @JvmStatic
  fun formatNumberSimple(number: Double): String {
    return formatNumber(NUMBER_DECIMAL_FORMAT_SIMPLE, roundNumber(number))
  }

  @JvmStatic
  fun formatNumber(number: Double): String {
    return formatNumber(NUMBER_DECIMAL_FORMAT, roundNumber(number))
  }

  @JvmStatic
  fun formatNumber(decimalFormat: DecimalFormat, number: Double): String {
    return decimalFormat.format(number)
  }

  @JvmStatic
  fun roundNumber(number: Float): Double {
    return (number * 100).roundToLong() / 100.0
  }

  @JvmStatic
  fun roundNumber(number: Double): Double {
    return (number * 100).roundToLong() / 100.0
  }
}